postgres/datadog_checks/postgres/statement_samples.py

```diff
@@ -469,3 +474,19 @@ def _explain_pg_stat_activity(self, rows):
                     1,
                     tags=self._tags + ["error:collect-plan-for-statement-crash"],
                 )
+
+    def _get_query_max_text_size(self):
+        with self._get_db(self._config.dbname).cursor(cursor_factory=psycopg2.extras.DictCursor) as cursor:
+            self._log.debug("Running query [%s]", MAX_ACTIVITY_QUERY_SIZE_QUERY)
+            cursor.execute(MAX_ACTIVITY_QUERY_SIZE_QUERY)
+            row = cursor.fetchone()
+            return int(row['setting'])
+
+    @staticmethod
+    def _is_statement_truncated(max_query_size, row):
+        # Compare the query length to the configured max query size to determine
+        # if the query has been truncated. Note that the length of a truncated statement
+        # is one less than the value of 'track_activity_query_size'. One caveat is that if a statement
+        # happens to have *exactly* the same length as the max configured but isn't actually truncated, this
+        # would falsely report it as a truncated statement
+        return sys.getsizeof(row['query']) >= max_query_size - 1
```

Comment:
Does this hold true for queries with multi-byte unicode strings? It would be good to add some tests on that  as well.

Reply:
Good point. Given postgres's documentation, I think counting bytes is right but I should add tests to validate.